components/compositing/compositor.rs

```diff
@@ -1156,14 +1160,34 @@ impl<Window: WindowMethods> IOCompositor<Window> {
         // Batch up all scroll events into one, or else we'll do way too much painting.
         let mut last_combined_event: Option<ScrollZoomEvent> = None;
         for scroll_event in self.pending_scroll_zoom_events.drain(..) {
-            let this_delta = scroll_event.delta;
             let this_cursor = scroll_event.cursor;
+
+            let this_delta = match scroll_event.scroll_location {
+                ScrollLocation::Delta(delta) => delta,
+                _ => {
```

Comment:
I'd prefer we remove the _ here and explicitly specify the Home/End enum values. That way, if/when new values are added to the enum, this becomes a compile error until they are accounted for here.